### PR TITLE
Move shared-resources and separate undo-redo plugin

### PR DESCRIPTION
**Pull request by saimonsaret**
1) Move shared-recources codebase to editor-service 
2) Split codebase into editor and 2dmodel 
3) Create additional grunt tasks for compiling editor and 2dmodel cores
4) Create plugin controller, which controls creation and usage of plugins
5) Separate undo-redo plugin and gestures plugin, project can be run without them
Ticket #127180365

### CODE REVIEW

**Comment by TanVD** [comment]
Let's merge master in this branch in order to have fast-forward merge.

**Comment by TanVD** [comment]
GeneralConstants should contain maven parameters in order to parametrize whole TS app. (Please, merge master and resolve conflicts)

File: editor-service/src/main/webapp/app/editor/interfaces/diagramFull.d.ts [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+
+    logical: Map<Property>;
+    graphical: Map<Property>;
+
+    constructor(logical: Map<Property>, graphical: Map<Property>);
+
+}
+
+declare class Link implements DiagramElement {
+
```
Comment (jzuken) [inline editor-service/src/main/webapp/app/editor/interfaces/diagramFull.d.ts]: these empty lines look ugly. I suggest we don't insert them

File: editor-service/src/main/webapp/WEB-INF/views/include/scripts.jsp [hunk 2]
```diff
@@ -12,10 +12,9 @@
...
-    <script src="<c:url value='/resources/js/compiled/utils.js' />"></script>
-    <script src="<c:url value='/resources/js/compiled/root.js' />"></script>
-    <script src="<c:url value='/resources/js/compiled/diagram-core.js' />"></script>
-    <script src="<c:url value='/resources/js/compiled/two-d-model-core.js' />"></script>
+    <script src="<c:url value='/resources/js/compiled/editorCore.js' />"></script>
+    <script src="<c:url value='/resources/js/compiled/twoDModelCore.js' />"></script>
     <script src="<c:url value='/resources/js/compiled/editor.js' />"></script>
+    <script src="<c:url value='/resources/js/compiled/undoRedoPlugin.js' />"></script>
     <script src="<c:url value='/resources/js/compiled/interpreter.js' />"></script>
     <script src="<c:url value='/resources/js/compiled/two-d-model-robots.js' />"></script>
```
Comment (TanVD) [inline editor-service/src/main/webapp/WEB-INF/views/include/scripts.jsp]: Newline... You can modify and use script from here [StackOverflow question](http://stackoverflow.com/questions/5922696/add-missing-newlines-in-multiple-files) and add newlines where it is necessary automatically.

**Comment by TanVD** [comment]
1. Add reference pack file for blocks
2. Rename createFromDao methods in accordance with current architecture of program

**Pull request closed without merging**